javascript/ql/src/semmle/javascript/security/dataflow/ClientSideUrlRedirect.qll

```diff
@@ -35,24 +43,37 @@ module ClientSideUrlRedirect {
       source instanceof Source
     }
 
+    override predicate isSource(DataFlow::Node source, DataFlow::FlowLabel lbl) {
+      source instanceof RemoteFlowSource and
+      lbl = DataFlow::FlowLabel::taint()
```

Comment:
Why does `RemoteFlowSource` start out as being of kind "taint"? I would have expected "data".

I see that the rest of the configuration expects "taint", but it should be able to handle both "data" and "taint" IMO.

Reply:
Yes, "data" would be better. I think the original reason for making it "taint" was that `isSink` expects "taint", but we should instead make that accept "data" as well.